backend/controller/internal/dal/events.go

```diff
@@ -253,6 +253,9 @@ func (d *DAL) QueryEvents(ctx context.Context, filters ...EventFilter) ([]Event,
 	}
 
 	q += " ORDER BY time_stamp ASC"
+	if limit != nil {
+		q += fmt.Sprintf(" LIMIT %d", *limit)
+	}
```

Comment:
Is this the right approach to this? I mostly needed limit for pagination. I could also add a Filter but it seemed like it might be more clear and outside of the filter flow, but curious what you think @alecthomas

Reply:
Looks fine to me, except for the use of `*int`

Reply:
Should I go with `int` instead and just check for `limit > 0`? Or use the `types.Option`?